src/python/Utils/Utilities.py

```diff
@@ -224,6 +227,27 @@ def decodeBytesToUnicode(value, errors="strict"):
     return value
 
 
+def encodeToString(value, errors="strict"):
+    """
+    This function encodes a sequence of unicode code points, or a sequence
+    of bytes (bytes type) to a sequence of byte strings under the str type.
+
+    py2:
+    - "errors" can be: "strict", "ignore", "replace",
+    - ref: https://docs.python.org/2/howto/unicode.html#the-unicode-type
+    py3:
+    - "errors" can be: "strict", "ignore", "replace", "backslashreplace"
+    - ref: https://docs.python.org/3/howto/unicode.html#the-string-type
+    """
+    if PY3:
+        if isinstance(value, bytes):
+            return str(value, 'utf-8', errors)
+    else:
+        # then encode either bytes / str / unicode to string
+        return value.encode('utf-8', errors)
+    return value
```

Comment:
We are importing `from builtins import str, bytes`, so in this file `str` and `bytes` are the same as in python3.
- `str` == unicode string
- `bytes` == bytes string

The purpose is safely encoding a unicode string to bytes, using the python3 naming: 

```suggestion
    if isinstance(value, str):
        return value.encode('utf-8', errors) # or bytes(value, 'utf-8', errors)
    return value
```

Reply:
I think this might be a problem in python3:
```
Python 3.8.2 (default, Nov 25 2020, 10:58:42) 
In [1]: from builtins import str, bytes

In [2]: mystr = 'alan'

In [3]: mybytes = b'alan'

In [4]: mystr == mybytes
Out[4]: False

In [5]: type(mystr)
Out[5]: str

In [6]: type(mybytes)
Out[6]: bytes
```

or am I missing something?

Reply:
No, this is correct! But `str` in py3 means unicode, while `str` in py2 means bytes.

if you run your function in py2, you would convert unicode to bytes ("encode"), because we are importing `str` from `builtins`. In py3, if  `value` is bytes you would convert it to unicode, else you would try to "encode" it, which means converting it to bytes, whatever the type of `value` is, without making sure that `value` has an `.encode()` attribute/member.